scripts/singleton/get-deploy-data.js

```diff
@@ -0,0 +1,34 @@
+const { lstatSync, readdirSync } = require('fs')
+const { join } = require('path')
+
+const ethTx = require('ethereumjs-tx');
+const ethUtils = require('ethereumjs-util');
+
+function toHexString(buffer) {
+  return '0x' + buffer.toString('hex')
+}
+
+const bin = require(`../../singleton/RelayHub.flattened.json`)
+  .contracts['singleton/RelayHub.flattened.sol:RelayHub']
+  .bin;
+
+const tx = new ethTx({
+  nonce: 0,
+  data: '0x' + bin,
+  value: 0,
+  gasPrice: 100000000000, /// 100 gigawei
+  gasLimit: 8000000,
+  v: 27,
+  r: '0x1613161316131613161316131613161316131613161316131613161316131613',
+  s: '0x1613161316131613161316131613161316131613161316131613161316131613'
+});
+
+const deployer = tx.getSenderAddress();
```

Comment:
Note that this may fail about half of the time, whenever the code of the contract changes, since not all synthetic signatures are valid. We may need to add a loop to either make a change in the contract or use a slightly different signature if it fails.

Reply:
I thought about this, but couldn't make it fail after running it over a thousand times with small changes to the source, so I figured there was something wrong with my understanding of ECDSA.

Reply:
Maybe it's related to choosing `v=27`? Anyway, if it doesn't fail, then it's good!

Reply:
Googled it. The ECDSA signature represents a point on a curve, so if this value is valid, changing the signed data won't invalidate it.